scripts/rctGCS.py

```diff
@@ -1176,53 +2064,587 @@ def __createWidgets(self):
+    def __cacheMap(self):
+        '''
+        Inner function to facilitate map caching
+        '''
+        if self.isWebMap:
+            if (self.mapWidget.toolRect.rectangle() == None):
+                msg = QMessageBox()
+                msg.setText("No specified area to cache!")
+                msg.setWindowTitle("Alert")
+                msg.setInformativeText("Use the rect tool to choose an area on the map to cache")
+                msg.setIcon(QMessageBox.Critical)
+                msg.exec_()
+                self.mapWidget.rect()
+            else:
+                cacheThread = Thread(target=self.mapWidget.cacheMap)
+                cacheThread.start()
+                self.mapWidget.canvas.refresh()
+        else:
+            print("alert")
 
-    def __loadMapFile(self):
-
-        filename = askopenfilename()
-        print(filename)
+    def setMap(self, mapWidg: MapWidget, isWebMap):
+        '''
+        Function to set the MapWidget that this object will use
+        Args:
+            mapWidg: A MapWidget object
+            isWebMap: A boolean indicating whether or not the mapWidget 
+                      is a WebMap
+        '''
+        self.isWebMap = isWebMap
+        self.mapWidget = mapWidg
+        
+        self.btn_cacheMap.setEnabled(isWebMap)
+        
+    def estDistance(self, coord):
+        '''
+        An inner function to display the distance from the 
+        current estimate point to the ground truth
+        Args:
+            coord: A tuple of float values indicating an EPSG:4326 lat/Long 
+                   coordinate pair
+            stale: A boolean
+            res: The residuals vector
+        '''
+        lat1 = coord[0]
+        lon1 = coord[1]
+        lat2 = 32.885889
+        lon2 = -117.234028
 
-        self.__clearMap()
+        
+        dist = self.distance(lat1, lat2, lon1, lon2)
+       
+        
+        d = '%.3f'%(dist)
 
-        fig = plt.figure(figsize=(5,5))
+        self.lbl_dist.setText(d + '(m.)')
         
-        my_image = georaster.SingleBandRaster(filename, load_data=False)
+    def distance(self, lat1, lat2, lon1, lon2): 
+        '''
+        Helper function to calculate distance
+        Args:
+            lat1: float value indicating the lat value of a point
+            lat2: float value indicating the lat value of a second point
+            lon1: float value indicating the long value of a point
+            lon2: float value indicating the long value of a second point
+        '''
+        lon1 = radians(lon1) 
+        lon2 = radians(lon2) 
+        lat1 = radians(lat1) 
+        lat2 = radians(lat2) 
+           
+        # Haversine formula  
+        dlon = lon2 - lon1  
+        dlat = lat2 - lat1 
+        a = sin(dlat / 2)**2 + cos(lat1) * cos(lat2) * sin(dlon / 2)**2
+      
+        c = 2 * asin(sqrt(a))  
+         
+        # Radius of earth in kilometers. Use 3956 for miles 
+        r = 6371
+           
+        return(c * r * 1000)
```

Comment:
Recommend to convert to decimal degrees to UTM, then do the distance function since UTM is in meters